app/modules/encounters/resources.py

```diff
@@ -0,0 +1,115 @@
+# encoding: utf-8
+# pylint: disable=bad-continuation
+"""
+RESTful API Encounters resources
+--------------------------
+"""
+
+import logging
+
+from flask_login import current_user
+from flask_restplus_patched import Resource
+from flask_restplus._http import HTTPStatus
+
+from app.extensions import db
+from app.extensions.api import Namespace
+from app.extensions.api.parameters import PaginationParameters
+from app.modules.users import permissions
+
+
+from . import parameters, schemas
+from .models import Encounter
+
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+api = Namespace('encounters', description="Encounters")  # pylint: disable=invalid-name
+
+
+@api.route('/')
+@api.login_required(oauth_scopes=['encounters:read'])
+class Encounters(Resource):
+    """
+    Manipulations with Encounters.
+    """
+
+    @api.parameters(PaginationParameters())
+    @api.response(schemas.BaseEncounterSchema(many=True))
+    def get(self, args):
+        """
+        List of Encounter.
+
+        Returns a list of Encounter starting from ``offset`` limited by ``limit``
+        parameter.
+        """
+        return Encounter.query.offset(args['offset']).limit(args['limit'])
+
+    @api.login_required(oauth_scopes=['encounters:write'])
+    @api.parameters(parameters.CreateEncounterParameters())
+    @api.response(schemas.DetailedEncounterSchema())
+    @api.response(code=HTTPStatus.CONFLICT)
+    def post(self, args):
+        """
+        Create a new instance of Encounter.
+        """
```

Comment:
I think this endpoint (and similar endpoints for other entities) need to call `self.view()` to update the timestamp, correct?

Reply:
that its correct